hal/ia32/memory.c

```diff
@@ -0,0 +1,108 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system loader
+ *
+ * Memory map
+ *
+ * Copyright 2012, 2020, 2021 Phoenix Systems
+ * Copyright 2001, 2005 Pawel Pisarczyk
+ * Author: Pawel Pisarczyk, Lukasz Kosinski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <hal/hal.h>
+#include <lib/errno.h>
+#include <syspage.h>
+
+
+/* Flushes 8042 keyboard controller buffers */
+static inline void memory_empty8042(void)
+{
+	unsigned char status;
+
+	do {
+		/* Discard input data */
+		if ((status = hal_inb((void *)0x64)) & 0x01) {
```

Comment:
This got me thinking - should hal_inb, hal_outb etc. take void * as an agument? Port address is not a pointer - it's a different memory space altogether. Maybe port address should be of integer type (unsigned short for example) or have it's own type defined?

Reply:
I agree that we could change port address type to `unsigned short` or `u16` (e.g. it's like that in Linux) (`void *` is also used as IO port type in kernel).
As a bigger change I'd like to leave it for a separate PR.